Assets/MixedRealityToolkit.Extensions/SceneTransitionService/ISceneTransitionService.cs

```diff
@@ -0,0 +1,133 @@
+
+using Microsoft.MixedReality.Toolkit.UI;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.SceneTransitions
+{
+    public interface ISceneTransitionService : IMixedRealityExtensionService
+    {
+        /// <summary>
+        /// Called when transition starts.
+        /// This is called at the beginning of a transition, not at the beginning of a scene load.
+        /// For scene load events, use the Scene System.
+        /// </summary>
+        Action OnTransitionStarted { get; set; }
+
+        /// <summary>
+        /// Called when transition ends.
+        /// This is called at the end of a transition, not at the end of a scene load.
+        /// For scene load events, use the Scene System.
+        /// </summary>
+        Action OnTransitionCompleted { get; set; }
+
+        /// <summary>
+        /// Whether to use a fade color during transitions.
+        /// </summary>
+        bool UseFadeColor { get; set; }
+
+        /// <summary>
+        /// The color to use when fading out.
+        /// </summary>
+        Color FadeColor { get; set; }
+
+        /// <summary>
+        /// Time in seconds for fade in to complete.
+        /// </summary>
+        float FadeInTime { get; set; }
+        
+        /// <summary>
+        /// Time in seconds for fade out to complete.
+        /// </summary>
+        float FadeOutTime { get; set; }
+
+        /// <summary>
+        /// Which cameras to target when fading.
+        /// </summary>
+        CameraFaderTargets FadeTargets { get; set; }
+
+        /// <summary>
+        /// True when a scene transition is in progress.
+        /// </summary>
+        bool TransitionInProgress { get; }
+
+        /// <summary>
+        /// From 0-1
+        /// </summary>
+        float TransitionProgress { get; }
+
+        /// <summary>
+        /// Fades out, enables progress indicator, execute scene operations in order, disables progress indicator, fades back in
+        /// </summary>
+        /// <param name="sceneOperations">A set of tasks from the Scene System.</param>
+        /// <param name="progressIndicator">If null, default progress indicator prefab will be used (or none if default is disabled in profile)</param>
+        /// <returns></returns>
+        Task DoSceneTransition(IEnumerable<Func<Task>> sceneOperations, IProgressIndicator progressIndicator = null);
+
+        /// <summary>
+        /// Fades out, enables progress indicator, executes scene op 1, executes scene op 2, disables progress indicator, fades back in
+        /// </summary>
+        /// <param name="sceneOp1"></param>
+        /// <param name="sceneOp2"></param>
+        /// <param name="progressIndicator"></param>
+        /// <returns></returns>
+        Task DoSceneTransition(Func<Task> sceneOp1, Func<Task> sceneOp2, IProgressIndicator progressIndicator = null);
+
```

Comment:
This is just a convenience version of the thing above right? It feels like you could just do DoSceneTransition({operation1, operation2} and it would be fairly straightforward as well).

It's just a larger API surface to maintain going forward and I'm curious if this means the next thing is a version that takes three functions (and where we draw the line - maybe it's just at two)

Reply:
Yes it's just convenience. I found I was writing a lot of boilerplate for simple transitions. With more complex transitions it didn't seem as problematic. I think we can draw the line at 2.